tests/conftest.py

```diff
@@ -15,3 +23,45 @@ def vcr_config():
     return {
         "filter_headers": [CROSSREF_HEADER_KEY, SEMANTIC_SCHOLAR_HEADER_KEY],
     }
+
+
+@pytest.fixture()
+def unique_tmp_subfolder(tmp_path):
+    yield tmp_path
+    # Cleanup after the test
+    if tmp_path.exists():
+        shutil.rmtree(tmp_path, ignore_errors=True)
+
+
+@pytest.fixture()
+def agent_module_dir(unique_tmp_subfolder, monkeypatch):
+    """Set up a unique temporary folder for the agent module."""
+    monkeypatch.setenv("PQA_HOME", str(unique_tmp_subfolder))
+    importlib.reload(paperqa.agents)
```

Comment:
Okay reading this it comes off as a bit hacky. I am left wondering things like:
- Why is `agent_module_dir` not just a static filepath to a Python module made by `pathlib.Path(__file__).parent`?
- How does `PQA_HOME` relates to `agent_module_dir`?
- Will `importlib.reload` have implications with multiprocessing or fixture ordering?

Also, are you open to not using `monkeypatch.setenv` here? Imo it's more intuitive if one `yield`s from within a context manager

Reply:
ok cutting the reload; it's not needed anymore anyway.

-  The name is bad, I'm changing this to `agent_home_dir` -- this is where the indices and project data are generated. It was named that way before since it's the directory to be used by the module to store data.
- not married to using monkeypatch, I can use patch w a yield.